RecoPPS/RPixEfficiencyTools/README.md

```diff
@@ -8,4 +52,57 @@ Instruction on how to setup and run efficiency suite in CMSSW_11_3_2 environment
+	2. **file list to analyse** Just create .dat file with paths you want to analyse.
+	```ls /eos/project-c/ctpps/subsystems/Pixel/RPixTracking/EfficiencyCalculation2018/ReRecoEras_CMSSW_10_6_10/EGamma/re-MINIAOD/ctpps_2018A_pixelefficiency_singleele200630_073503/0000/*.root | sed 's/^/file:/' > InputFiles/Era.dat```
+	3. **Set up config files** (.py files in python folder).  You just need make step 1 manually, all others values are correct. They are put here for debugging purposes.  
+		1. **MaxEvents** as test file is big just set maxEvents to smaller value ex. `100000`
+		2. **Global tag** Use `123X_dataRun2_v4` Global Tag in the python files.
+		3. **Process**  `eras.Run2_2018`
+		4. **Producer Tag** `CTPPSTestProtonReconstruction`
+		
+3. Run Efficiency analysis (referred as EA).
+	1. **Run EA (efficiency analysis) worker**
+		```cmsRun python/EfficiencyAnalysisDQMWorker_cfg.py sourceFileList=InputFiles/Era.dat outputFileName=ea_tmp.root bunchSelection=NoSelection```
+	2. **Run EA (efficiency analysis) harvester** To run EA you need to specify 5 paramteres
+		1. **inputFileName** .root output file of EA Worker
+		2. **outputDirectoryPath** where to save output of root file 
+		3. **workflow** ex. efficiency, timing resolution
+		4. **dataPeriod** runNumber or lumisection ex. `315512`
+		5. **campaign** provides a unique naming for distinction between same workflow ex. `25_07_22`
+	```cmsRun python/EfficiencyAnalysisDQMHarvester_cfg.py inputFileName=ea_tmp.root outputDirectoryPath=OutputFiles workflow=randomNameWorkflow dataPeriod=dumbDataPeriod campaign=fooCampaignName```
+
+4. Run Reference Analysis (referred as RA)
+	1. **Run RA worker** 
+		1. **sourceFileList** self describing
+		2. **outputFileName** self describing 
+		3. **efficiencyFileName** .root file output of EA Harvester
+	```cmsRun python/ReferenceAnalysisDQMWorker_cfg.py sourceFileList=InputFiles/Era.dat outputFileName=tmp.root efficiencyFileName=OutputFiles/DQM_V0001_R000999999__fooCampaignName__randomNameWorkflow__dumbDataPeriod.root```
+
+	2. **Run RA harvester**
+
+	TODO: Probably outputDirectoryPath doesn't change the output path. 
+	```cmsRun python/ReferenceAnalysisDQMHarvester_cfg.py inputFileName=tmp.root outputDirectoryPath=OutputFiles```
+5. Verification
+	1. Open root file from RA Harvester *(note: More instruction in the `Onboarding section` of this readme)*
+	2. Check a few histograms. Ex.`h2RefinedTrackEfficiency_am0_*`, `h2TrackHitDistribution_*`
+	Below images of example histograms.
+	![](ReadmeResources/h2_RefinedTrackEfficiency_example_histogram.png)
+	![](ReadmeResources/h2_HitTrackDistribution_example_histogram.png)
+
+	3. You can limit data to `colz` (Set `draw option`) 
+
+# TODO - Hope it will be done in a finite time :)  
+1. [ ] Global tag is set in every python file separately - if you change a global tag you need to edit four files!!!

```

Comment:
A similar thing must be done for the era modifier --> just write it down as a TODO, then we'll think about it. It's a quick modification